go/libraries/doltcore/doltdb/commit_hooks.go

```diff
@@ -127,3 +184,69 @@ func (lh *LogHook) SetLogger(ctx context.Context, wr io.Writer) error {
 	lh.out = wr
 	return nil
 }
+
+func RunAsyncReplicationThreads(ctx context.Context, wg *sync.WaitGroup, ch chan PushArg, destDB *DoltDB, tmpDir string, logger io.Writer) {
+	mu := &sync.Mutex{}
+	var newHeads = make(map[string]PushArg, asyncPushBufferSize)
+
+	// stop first go routine before second
+	newCtx, stop := context.WithCancel(context.Background())
+	wg.Add(1)
+	go func() error {
+		defer wg.Done()
+		for {
+			select {
+			case p, ok := <-ch:
+				if !ok {
+					return ctx.Err()
+				}
+				mu.Lock()
+				newHeads[p.ds.ID()] = p
+				mu.Unlock()
+			case <-ctx.Done():
+				stop()
+				return ctx.Err()
+			}
+		}
+	}()
+
+	wg.Add(1)
+	go func() error {
```

Comment:
same as above.  Either don't return the error, do something with it, or assign it to _.